File: solutions/c/pangram/1/pangram.c
#include "pangram.h"
#include <string.h>



bool is_pangram(const char *sentence){
    if(!sentence)
        return false;
    size_t len = strlen(sentence);
    if(0==len) 
        return false;
    bool can_be_pangram = true;
    for(int i=0; i<26; i++){
        bool contains_letter = false;
        for(size_t j=0; j<len; j++){
            if(sentence[j] == 'a' + i || sentence[j] == 'A' +i ){
                contains_letter = true;
                break;
            }
        }
        if(!(can_be_pangram = contains_letter && can_be_pangram))
            break;        
    }   
    return can_be_pangram;    
}

File: solutions/fortran/pangram/1/pangram.f90
module pangram
  implicit none
contains

  logical function has_letter(sentence, no)
    character(*) :: sentence
    integer :: no
    character(26) :: lowercase, uppercase
    integer :: i 
    character :: cl, cu
    lowercase = "abcdefghijklmnopqrstuvwxyz"
    uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    has_letter = .false.
    cl = lowercase(no:no) 
    cu = uppercase(no:no)
    do i = 1,len(sentence)
       has_letter = has_letter.or.(sentence(i:i) == cl).or.(sentence(i:i) == cu)
    end do
  end function has_letter

  logical function is_pangram(sentence) 
    character(*) :: sentence
    integer :: lts 
    is_pangram = .true.
    do lts = 1,26
        is_pangram = is_pangram.and.has_letter(trim(sentence), lts)
    end do
   end function is_pangram

end module pangram
